packages/core/src/integrations/mcp-server/transport.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Transport layer instrumentation for MCP server
+ * Handles message interception and response correlation
+ */
+
+import { getIsolationScope, withIsolationScope } from '../../currentScopes';
+import { startInactiveSpan, withActiveSpan } from '../../tracing';
+import { fill } from '../../utils/object';
+import { cleanupAllPendingSpans, completeSpanWithResults, storeSpanForRequest } from './correlation';
+import { captureError } from './errorCapture';
+import { buildMcpServerSpanConfig, createMcpNotificationSpan, createMcpOutgoingNotificationSpan } from './spans';
+import type { ExtraHandlerData, MCPTransport } from './types';
+import { isJsonRpcNotification, isJsonRpcRequest, isJsonRpcResponse } from './validation';
+
+/**
+ * Wraps transport.onmessage to create spans for incoming messages
+ */
+export function wrapTransportOnMessage(transport: MCPTransport): void {
+  if (transport.onmessage) {
+    fill(transport, 'onmessage', originalOnMessage => {
+      return function (this: MCPTransport, jsonRpcMessage: unknown, extra?: unknown) {
+        if (isJsonRpcRequest(jsonRpcMessage)) {
+          const messageTyped = jsonRpcMessage as { method: string; id: string | number };
+
+          // Create isolation scope for this request (standard Sentry pattern)
```

Comment:
m: Instead of inline comments like this, I would prefer if we added notes in the jsdoc to document behaviour. Right now this comment is a bit redundant with the code.

Ditto with other instances of this.